src/app/components/Pokemon.tsx

```diff
@@ -0,0 +1,48 @@
+'use client';
+
+import Link from 'next/link';
+import type { Pokemon, PokemonSprites, PokemonType } from '~/libs/poke/dto/pokemon';
+import { useState } from 'react';
+
+interface PokemonProps {
+  pokemon: Pokemon
+}
+
+function PokemonRN(props: PokemonProps) {
+  const { pokemon } = props;
+
+  const [IsTurned, setIsTurned] = useState<boolean>(false);
+
+  const sprite_name = (!IsTurned ? "front" : "back") + "_default";
+
+  return (
+    <div className="flex flex-col justify-center h-full px-8 ">
+      <Link className="text-gray-300 underline mb-4" href="/">
+        Home
+      </Link>
+      <h1 className="text-7xl font-bold w-full text-center capitalize">{pokemon.name}</h1>
+
+        <div className='w-[80%] m-auto flex flex-col justify-center items-center'>
+          <img src={pokemon.sprites[sprite_name]} className='w-full max-w-[400px]' />
+          
+
+          <button onClick={() => setIsTurned(!IsTurned)} role='button' className='cursor-pointer'>
```

Comment:
Risque de stale state, je te recommande cette article pour comprendre et corriger cela
https://medium.com/@shanakaprince/understanding-stale-state-in-react-and-how-to-avoid-it-a74cc4655c43